### ISSUE TITLE
Test failure for SCTP

### ISSUE THREAD

**Issue by Jille** [Jille/rufs#82]
This test failed during an unrelated PR.

https://github.com/sgielen/rufs/runs/5082942193

```
=== RUN   TestStuff
2022/02/06 10:32:01 OpenStream(5) when dialing to "127.0.0.1:40537"
2022/02/06 10:32:01 OpenStream(4) when dialing to "127.0.0.1:37910"
2022/02/06 10:32:01 AcceptStream returned 5 from 127.0.0.1:37910
2022/02/06 10:32:01 AcceptStream returned 4 from 127.0.0.1:40537
sctp ERROR: 2022/02/06 10:32:03 [0xc0000ee1c0] stream 5 not found)
    udp_test.go:90: Received buffers are not equal to sent ones
--- FAIL: TestStuff (2.34s)
```

**Comment by Jille** [comment]
I reproduced this on yesterday's `master`. Sometimes it does succeed, sometimes it fails like this.

```
go: downloading github.com/pion/sctp v1.8.2
=== RUN   TestStuff
2024/08/17 13:07:00 OpenStream(5) when dialing to "127.0.0.1:55086"
2024/08/17 13:07:00 OpenStream(4) when dialing to "127.0.0.1:65253"
2024/08/17 13:07:00 AcceptStream returned 4 from 127.0.0.1:55086
2024/08/17 13:07:00 AcceptStream returned 5 from 127.0.0.1:65253
sctp ERROR: 2024/08/17 13:07:02 [0x1400013c000] stream 4 not found)
    udp_test.go:90: Received buffers are not equal to sent ones
```

Current master however, just hangs forever on `eg.Wait` and never succeeds.

**Comment by Jille** [comment]
Fixed with `38aaf9d57b40ff3400693977ce8b58aa16f47106`
